src/@next/ComboBox/components/OptionList/OptionList.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useComboboxTextInput } from '../TextInput/TextInputContext';
+import { NoOptionList, NoOptionListProps } from './Option';
+import { ComboboxOptionListContext } from './OptionListContext';
+import { OptionListContainer, StyledOptionList } from './OptionListStyle';
+
+export interface OptionListProps extends NoOptionListProps {
+  children: React.ReactNode;
+  isEmpty?: boolean;
+  onSelect?(value: string): void;
+}
+
+export const OptionList = ({
+  children,
+  isEmpty,
+  noOptionsMessage,
+  onSelect,
+}: OptionListProps) => {
+  const textInputContext = useComboboxTextInput();
+  const { textInputWidth } = textInputContext;
+  const hasChildren = !isEmpty;
+
+  const onOptionSelect = (value: string) => {
+    if (onSelect) onSelect(value);
```

Comment:
can we have curly bracket or optional chaining, I think we should set ESLint rule for this